amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```diff
@@ -203,6 +209,31 @@ private void updateLeasesForChildShards()
         log.info("Shard {}: Updated current lease {} with child shard information: {}", shardInfo.shardId(), currentLease.leaseKey(), childShardIds);
     }
 
+    private void garbageCollectLeaseForParentShard() throws DependencyException, ProvisionedThroughputException, InvalidStateException {
+       final Lease parentShardLease = leaseCoordinator.getCurrentlyHeldLease(leaseKeyProvider.apply(shardInfo));
+
+        if (parentShardLease.childShardIds().isEmpty()) {
```

Comment:
Could use `CollectionUtils.isNullOrEmpty` just in case childShardIds is null

Reply:
`childShardIds` will never be null with the current Lease implementation, but i'll add this anyways incase this ever changes in the future.